internal/engine/experiment/torsf/torsf.go

```diff
@@ -31,6 +43,15 @@ type TestKeys struct {
 
 	// Failure contains the failure string or nil.
 	Failure *string `json:"failure"`
+
+	// PersistentDatadir indicates whether we're using a persistent tor datadir.
+	PersistentDatadir bool `json:"persistent_datadir"`
```

Comment:
Do we perhaps also want to include a timestamp (maybe rounded to the day) of when the datadir was last modified and if it's empty? I am thinking it would be helpful to know if your datadir actually has data in it or if the data is stale, hence needing to fetch descriptor information.

Reply:
I don't think it's necessary. The logs I'm including already clearly show whether the cache is cold or not. I think it is better to rely on parsing the logs than adding more complexity on the probe. Tor is known for stable output messages AFAICT, and therefore I would rely on that rather than adding more possible fragile code to check whether the cache is cold. (It may in principle be that the cache does not seem cold but actually it is cold: the time of the last modification is just a proxy and we can tell whether it was cold or not very reliably if we observe what tor had to do.)

Reply:
I have documented into the related ooni/spec changes in https://github.com/ooni/spec/pull/231 the difference in terms of tor logs between a bootstrap with a stale cache and one with a valid cache. Because the different is quite obvious, I'd rather just leave the pipeline to post-process this information and decide about the state of the cache.

(It must also be said that, interestingly, when I was producing data for updating https://github.com/ooni/spec/pull/231 I saw a bootstrap with stale cache converging in 67 s and one with a good/fresh cache taking ~300 s.)